app/create_app.py

```diff
@@ -114,22 +114,26 @@ def inject_global_constants():
         )
 
     @flask_app.context_processor
-    def inject_service_name():
-        fund = None
-        if request.view_args or request.args or request.form:
-            try:
-                fund = find_fund_in_request()
-            except Exception as e:  # noqa
-                current_app.logger.warning(
-                    f"""Exception: {e}, occured when trying to
-                    reach url: {request.url}, with view_args:
-                    {request.view_args}, and args: {request.args}"""
-                )
-        if fund:
-            service_title = gettext("Apply for") + " " + fund.title
-        else:
-            service_title = gettext("Access Funding")
-        return dict(service_title=service_title)
+    def utility_processor():
+        def inject_service_name():
+            fund = None
+            if request.view_args or request.args or request.form:
+                try:
+                    fund = find_fund_in_request()
+                except Exception as e:  # noqa
+                    current_app.logger.warning(
+                        f"""Exception: {e}, occured when trying to
+                        reach url: {request.url}, with view_args:
+                        {request.view_args}, and args: {request.args}"""
+                    )
+            if fund:
+                service_title = gettext("Apply for") + " " + fund.title
+            else:
+                service_title = gettext("Access Funding")
+
+            return service_title
+
+        return dict(service_title=inject_service_name())
```

Comment:
This still just gets executed eagerly/immediately, doesn't it?

Reply:
I have to fix this. By invoking the function there, it makes it act exactly as before (on every request). 
Instead, I should just pass the function. 

Had it like that initially, but then changed it without checking again.

Reply:
> I have to fix this. By invoking the function there, it makes it act exactly as before (on every request). Instead, I should just pass the function.

This reads exactly like chatgpt 😂

Reply:
I have achieved my true potential. expandingbrain.jpg

Reply:
Fixed it to invoke the function in the template.